src/foam/lib/PermissionPredicate.js

```diff
@@ -27,11 +27,18 @@ foam.CLASS({
     {
       name: 'f',
       javaCode: `
-        X x = (X) obj;
-        AuthService auth = (AuthService) x.get("auth");
-        return auth.check(x, permission_);
+        // java side is only needed for rules which is handled in the RulePermissionPredicate
+        throw new RuntimeException( new UnsupportedOperationException() );
       `,
-      code: async function() {
+      code: async function(objId) {    
```

Comment:
Does this work? Can predicates be async?

Reply:
Yeah it works. However, I can change it to a simple thenner instead since theres no looping of promises, don't really need async in this case